O/oneAPI_Support/build_tarballs.jl

```diff
@@ -91,8 +125,11 @@ CMAKE_FLAGS=()
 CMAKE_FLAGS+=(-DCMAKE_BUILD_TYPE=RelWithDebInfo)
 # Install things into $prefix
 CMAKE_FLAGS+=(-DCMAKE_INSTALL_PREFIX=${prefix})
+# Search for libraries in the prefix
+# XXX: why is this needed?
+CMAKE_FLAGS+=(-DCMAKE_SHARED_LINKER_FLAGS="-L${libdir}")
```

Comment:
That's probably an indication of something weird in their build system, we usually don't need to do that manually

Reply:
We're maintaining the CMakeLists.jl though, and I don't think I'm doing anything outrageous: https://github.com/JuliaGPU/oneAPI.jl/blob/master/deps/CMakeLists.txt

Reply:
Ok, after looking into this a little bit I believe the problem is that you're using Clang, because GCC would find the libraries automatically without `-L${libdir}`:
```console
sandbox:${WORKSPACE}/srcdir/oneAPI.jl/deps/build # gcc -v -Wl,-t --sysroot=/opt/x86_64-linux-gnu/x86_64-linux-gnu/sys-root/ -fPIC -O2 -g -DNDEBUG   -shared -Wl,-soname,liboneapi_support.so -o liboneapi_support.so CMakeFiles/oneapi_support.dir/src/sycl.cpp.o CMakeFiles/oneapi_support.dir/src/onemkl.cpp.o  -lmkl_sycl  -lmkl_intel_ilp64  -lmkl_sequential  -lmkl_core  -lsycl  -lOpenCL  -lpi_level_zero
Using built-in specs.
COLLECT_GCC=/opt/x86_64-linux-gnu/bin/x86_64-linux-gnu-gcc
COLLECT_LTO_WRAPPER=/opt/x86_64-linux-gnu/bin/../libexec/gcc/x86_64-linux-gnu/8.1.0/lto-wrapper
Target: x86_64-linux-gnu
Configured with: /workspace/srcdir/gcc-8.1.0/configure --prefix=/workspace/destdir --target=x86_64-linux-gnu --host=x86_64-linux-musl --build=x86_64-linux-musl --disable-multilib --disable-werror --enable-shared --enable-host-shared --enable-threads=posix --with-sysroot=/workspace/destdir/x86_64-linux-gnu/sys-root --program-prefix=x86_64-linux-gnu- --disable-bootstrap --with-arch=x86-64 --enable-languages=c,c++,fortran,objc,obj-c++
Thread model: posix
gcc version 8.1.0 (GCC) 
COMPILER_PATH=/opt/x86_64-linux-gnu/bin/../libexec/gcc/x86_64-linux-gnu/8.1.0/:/opt/x86_64-linux-gnu/bin/../libexec/gcc/:/opt/x86_64-linux-gnu/bin/../lib/gcc/x86_64-linux-gnu/8.1.0/../../../../x86_64-linux-gnu/bin/
LIBRARY_PATH=/opt/x86_64-linux-gnu/bin/../lib/gcc/x86_64-linux-gnu/8.1.0/:/opt/x86_64-linux-gnu/bin/../lib/gcc/:/opt/x86_64-linux-gnu/bin/../lib/gcc/x86_64-linux-gnu/8.1.0/../../../../x86_64-linux-gnu/lib/../lib64/:/opt/x86_64-linux-gnu/x86_64-linux-gnu/sys-root/lib/../lib64/:/opt/x86_64-linux-gnu/x86_64-linux-gnu/sys-root/usr/lib/../lib64/:/opt/x86_64-linux-gnu/bin/../lib/gcc/x86_64-linux-gnu/8.1.0/../../../../x86_64-linux-gnu/lib/:/opt/x86_64-linux-gnu/x86_64-linux-gnu/sys-root/lib/:/opt/x86_64-linux-gnu/x86_64-linux-gnu/sys-root/usr/lib/
COLLECT_GCC_OPTIONS='-D' '_GLIBCXX_USE_CXX11_ABI=1' '-frandom-seed=0x5d09198f' '-L/opt/x86_64-linux-gnu/x86_64-linux-gnu/lib64' '-march=x86-64' '-mtune=generic' '-v' '-fPIC' '-O2' '-g' '-D' 'NDEBUG' '-shared' '-o' 'liboneapi_support.so'
 /opt/x86_64-linux-gnu/bin/../libexec/gcc/x86_64-linux-gnu/8.1.0/collect2 -plugin /opt/x86_64-linux-gnu/bin/../libexec/gcc/x86_64-linux-gnu/8.1.0/liblto_plugin.so -plugin-opt=/opt/x86_64-linux-gnu/bin/../libexec/gcc/x86_64-linux-gnu/8.1.0/lto-wrapper -plugin-opt=-fresolution=/tmp/ccFmEenm.res -plugin-opt=-pass-through=-lgcc -plugin-opt=-pass-through=-lgcc_s -plugin-opt=-pass-through=-lc -plugin-opt=-pass-through=-lgcc -plugin-opt=-pass-through=-lgcc_s --sysroot=/opt/x86_64-linux-gnu/x86_64-linux-gnu/sys-root/ --eh-frame-hdr -m elf_x86_64 -shared -o liboneapi_support.so /opt/x86_64-linux-gnu/x86_64-linux-gnu/sys-root/usr/lib/../lib64/crti.o /opt/x86_64-linux-gnu/bin/../lib/gcc/x86_64-linux-gnu/8.1.0/crtbeginS.o -L/opt/x86_64-linux-gnu/x86_64-linux-gnu/lib64 -L/opt/x86_64-linux-gnu/bin/../lib/gcc/x86_64-linux-gnu/8.1.0 -L/opt/x86_64-linux-gnu/bin/../lib/gcc -L/opt/x86_64-linux-gnu/bin/../lib/gcc/x86_64-linux-gnu/8.1.0/../../../../x86_64-linux-gnu/lib/../lib64 -L/opt/x86_64-linux-gnu/x86_64-linux-gnu/sys-root/lib/../lib64 -L/opt/x86_64-linux-gnu/x86_64-linux-gnu/sys-root/usr/lib/../lib64 -L/opt/x86_64-linux-gnu/bin/../lib/gcc/x86_64-linux-gnu/8.1.0/../../../../x86_64-linux-gnu/lib -L/opt/x86_64-linux-gnu/x86_64-linux-gnu/sys-root/lib -L/opt/x86_64-linux-gnu/x86_64-linux-gnu/sys-root/usr/lib -rpath-link /opt/x86_64-linux-gnu/x86_64-linux-gnu/lib64 -t -soname liboneapi_support.so CMakeFiles/oneapi_support.dir/src/sycl.cpp.o CMakeFiles/oneapi_support.dir/src/onemkl.cpp.o -lmkl_sycl -lmkl_intel_ilp64 -lmkl_sequential -lmkl_core -lsycl -lOpenCL -lpi_level_zero -lgcc --as-needed -lgcc_s --no-as-needed -lc -lgcc --as-needed -lgcc_s --no-as-needed /opt/x86_64-linux-gnu/bin/../lib/gcc/x86_64-linux-gnu/8.1.0/crtendS.o /opt/x86_64-linux-gnu/x86_64-linux-gnu/sys-root/usr/lib/../lib64/crtn.o
/opt/x86_64-linux-gnu/bin/../lib/gcc/x86_64-linux-gnu/8.1.0/../../../../x86_64-linux-gnu/bin/ld: mode elf_x86_64
/opt/x86_64-linux-gnu/x86_64-linux-gnu/sys-root/usr/lib/../lib64/crti.o
/opt/x86_64-linux-gnu/bin/../lib/gcc/x86_64-linux-gnu/8.1.0/crtbeginS.o
CMakeFiles/oneapi_support.dir/src/sycl.cpp.o
CMakeFiles/oneapi_support.dir/src/onemkl.cpp.o
-lmkl_sycl_blas (/opt/x86_64-linux-gnu/x86_64-linux-gnu/sys-root//usr/local/lib/libmkl_sycl_blas.so)
-lmkl_sycl_lapack (/opt/x86_64-linux-gnu/x86_64-linux-gnu/sys-root//usr/local/lib/libmkl_sycl_lapack.so)
-lmkl_sycl_sparse (/opt/x86_64-linux-gnu/x86_64-linux-gnu/sys-root//usr/local/lib/libmkl_sycl_sparse.so)
-lmkl_sycl_dft (/opt/x86_64-linux-gnu/x86_64-linux-gnu/sys-root//usr/local/lib/libmkl_sycl_dft.so)
-lmkl_sycl_vm (/opt/x86_64-linux-gnu/x86_64-linux-gnu/sys-root//usr/local/lib/libmkl_sycl_vm.so)
-lmkl_sycl_rng (/opt/x86_64-linux-gnu/x86_64-linux-gnu/sys-root//usr/local/lib/libmkl_sycl_rng.so)
-lmkl_sycl_stats (/opt/x86_64-linux-gnu/x86_64-linux-gnu/sys-root//usr/local/lib/libmkl_sycl_stats.so)
-lmkl_sycl_data_fitting (/opt/x86_64-linux-gnu/x86_64-linux-gnu/sys-root//usr/local/lib/libmkl_sycl_data_fitting.so)
-lmkl_intel_ilp64 (/opt/x86_64-linux-gnu/x86_64-linux-gnu/sys-root//usr/local/lib/libmkl_intel_ilp64.so)
-lmkl_sequential (/opt/x86_64-linux-gnu/x86_64-linux-gnu/sys-root//usr/local/lib/libmkl_sequential.so)
-lmkl_core (/opt/x86_64-linux-gnu/x86_64-linux-gnu/sys-root//usr/local/lib/libmkl_core.so)
-lsycl (/opt/x86_64-linux-gnu/x86_64-linux-gnu/sys-root//usr/local/lib/libsycl.so)
-lOpenCL (/opt/x86_64-linux-gnu/x86_64-linux-gnu/sys-root//usr/local/lib/libOpenCL.so)
-lpi_level_zero (/opt/x86_64-linux-gnu/x86_64-linux-gnu/sys-root//usr/local/lib/libpi_level_zero.so)
libgcc_s.so.1 (/opt/x86_64-linux-gnu/x86_64-linux-gnu/lib64/libgcc_s.so.1)
/lib64/libc.so.6 (/opt/x86_64-linux-gnu/x86_64-linux-gnu/sys-root//lib64/libc.so.6)
/lib64/ld-linux-x86-64.so.2 (/opt/x86_64-linux-gnu/x86_64-linux-gnu/sys-root//lib64/ld-linux-x86-64.so.2)
/lib64/ld-linux-x86-64.so.2 (/opt/x86_64-linux-gnu/x86_64-linux-gnu/sys-root//lib64/ld-linux-x86-64.so.2)
libgcc_s.so.1 (/opt/x86_64-linux-gnu/x86_64-linux-gnu/lib64/libgcc_s.so.1)
/opt/x86_64-linux-gnu/bin/../lib/gcc/x86_64-linux-gnu/8.1.0/crtendS.o
/opt/x86_64-linux-gnu/x86_64-linux-gnu/sys-root/usr/lib/../lib64/crtn.o
COLLECT_GCC_OPTIONS='-D' '_GLIBCXX_USE_CXX11_ABI=1' '-frandom-seed=0x5d09198f' '-L/opt/x86_64-linux-gnu/x86_64-linux-gnu/lib64' '-march=x86-64' '-mtune=generic' '-v' '-fPIC' '-O2' '-g' '-D' 'NDEBUG' '-shared' '-o' 'liboneapi_support.so'
sandbox:${WORKSPACE}/srcdir/oneAPI.jl/deps/build # clang -v -Wl,-t --sysroot=/opt/x86_64-linux-gnu/x86_64-linux-gnu/sys-root/ -fPIC -O2 -g -DNDEBUG   -shared -Wl,-soname,liboneapi_support.so -o liboneapi_support.so CMakeFiles/oneapi_support.dir/src/sycl.cpp.o CMakeFiles/oneapi_support.dir/src/onemkl.cpp.o  -lmkl_sycl  -lmkl_intel_ilp64  -lmkl_sequential  -lmkl_core  -lsycl  -lOpenCL  -lpi_level_zero
clang version 16.0.6 (/home/gbaraldi/.julia/dev/BinaryBuilderBase/deps/downloads/clones/llvm-project.git-1df819a03ecf6890e3787b27bfd4f160aeeeeacd50a98d003be8b0893f11a9be 7cbf1a2591520c2491aa35339f227775f4d3adf6)
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /opt/x86_64-linux-musl/bin
Selected GCC installation: /opt/x86_64-linux-gnu/lib/gcc/x86_64-linux-gnu/8.1.0
Candidate multilib: .;@m64
Selected multilib: .;@m64
 "/opt/bin/x86_64-linux-gnu-libgfortran5-cxx11/x86_64-linux-gnu-ld.lld" --sysroot=/opt/x86_64-linux-gnu/x86_64-linux-gnu/sys-root/ -z now -z relro --hash-style=gnu --eh-frame-hdr -m elf_x86_64 -shared -o liboneapi_support.so /opt/x86_64-linux-gnu/x86_64-linux-gnu/sys-root/usr/lib/../lib64/crti.o /opt/x86_64-linux-gnu/lib/gcc/x86_64-linux-gnu/8.1.0/crtbeginS.o -L/opt/x86_64-linux-gnu/lib/gcc/opt/x86_64-linux-gnu/lib/gcc -L/opt/x86_64-linux-gnu/x86_64-linux-gnu/lib -L/opt/x86_64-linux-gnu/lib/gcc/x86_64-linux-gnu/8.1.0 -L/opt/x86_64-linux-gnu/x86_64-linux-gnu/sys-root/lib -L/opt/x86_64-linux-gnu/lib/gcc/x86_64-linux-gnu/8.1.0 -L/opt/x86_64-linux-gnu/lib/gcc/x86_64-linux-gnu/8.1.0/../../../../x86_64-linux-gnu/lib/../lib64 -L/opt/x86_64-linux-gnu/x86_64-linux-gnu/sys-root/lib/../lib64 -L/opt/x86_64-linux-gnu/x86_64-linux-gnu/sys-root/usr/lib/../lib64 -L/opt/x86_64-linux-gnu/lib/gcc/x86_64-linux-gnu/8.1.0/../../../../x86_64-linux-gnu/lib -L/opt/x86_64-linux-gnu/x86_64-linux-gnu/sys-root/lib -L/opt/x86_64-linux-gnu/x86_64-linux-gnu/sys-root/usr/lib -t -soname liboneapi_support.so CMakeFiles/oneapi_support.dir/src/sycl.cpp.o CMakeFiles/oneapi_support.dir/src/onemkl.cpp.o -lmkl_sycl -lmkl_intel_ilp64 -lmkl_sequential -lmkl_core -lsycl -lOpenCL -lpi_level_zero -lgcc --as-needed -lgcc_s --no-as-needed -lc -lgcc --as-needed -lgcc_s --no-as-needed /opt/x86_64-linux-gnu/lib/gcc/x86_64-linux-gnu/8.1.0/crtendS.o /opt/x86_64-linux-gnu/x86_64-linux-gnu/sys-root/usr/lib/../lib64/crtn.o
ld.lld: error: unable to find library -lmkl_sycl
ld.lld: error: unable to find library -lmkl_intel_ilp64
ld.lld: error: unable to find library -lmkl_sequential
ld.lld: error: unable to find library -lmkl_core
ld.lld: error: unable to find library -lsycl
ld.lld: error: unable to find library -lOpenCL
ld.lld: error: unable to find library -lpi_level_zero
clang-16: error: linker command failed with exit code 1 (use -v to see invocation)
sandbox:${WORKSPACE}/srcdir/oneAPI.jl/deps/build # clang -print-search-dirs
programs: =/opt/x86_64-linux-musl/bin:/opt/x86_64-linux-gnu/lib/gcc/x86_64-linux-gnu/8.1.0/../../../../x86_64-linux-gnu/bin
libraries: =/opt/x86_64-linux-musl/lib/clang/16:/opt/x86_64-linux-gnu/lib/gcc/x86_64-linux-gnu/8.1.0:/opt/x86_64-linux-gnu/lib/gcc/x86_64-linux-gnu/8.1.0/../../../../x86_64-linux-gnu/lib/../lib64:/opt/x86_64-linux-gnu/x86_64-linux-gnu/sys-root/lib/../lib64:/opt/x86_64-linux-gnu/x86_64-linux-gnu/sys-root/usr/lib/../lib64:/opt/x86_64-linux-gnu/lib/gcc/x86_64-linux-gnu/8.1.0/../../../../x86_64-linux-gnu/lib:/opt/x86_64-linux-gnu/x86_64-linux-gnu/sys-root/lib:/opt/x86_64-linux-gnu/x86_64-linux-gnu/sys-root/usr/lib
```
However what confuses me is that
* gcc and clang seem to have more or less the same search dirs for the libraries,
* gcc finds the libraries in `opt/x86_64-linux-gnu/x86_64-linux-gnu/sys-root//usr/local/lib/` (which is a symlink to `${libdir}`), but `gcc -v` doesn't list that dir in the search path :thinking: 

If the question is why you need to do this now, last build of this package was about one year ago, and clang changed a bit since then.  Maybe @gbaraldi has some ideas about what are the default search paths of new clang.